skglm/solvers/base.py

```diff
@@ -39,3 +57,45 @@ def solve(self, X, y, datafit, penalty, w_init, Xw_init):
         stop_crit : float
             Value of stopping criterion at convergence.
         """
+
+    def custom_compatibility_check(self, X, y, datafit, penalty):
+        """Ensure the solver is suited for the `datafit` + `penalty` problem.
+
+        This method includes extra checks to perform
+        aside from checking attributes compatibility.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples, n_features)
+            Training data.
+
+        y : array, shape (n_samples,)
```

Comment:
I agree with @Badr-MOUFAD here, even if not needed at the moment it's not too hard to see cases where this would happen, and an API change will be painful